content/articles/integrate-airtable-with-nextjs/index.md

```diff
@@ -0,0 +1,601 @@
+
+```js
+import { table, minifyItems } from '../../utils/Airtable'
+
+export default async (_req, res) => {
+	try {
+		const records = await table.select({}).firstPage()
+    const minfiedItems = minifyItems(records)
+		res.status(200).json(minfiedItems)
+	} catch (error) {
+		console.error(err)
+		res.status(500).json({ msg: 'Something went wrong! 😕' })
+	}
+}
+```
+
+### Create a new record
+To create new records, you can use the `create` method. It takes an array of up to 10 record objects. Each of the record objects should have `fields` key with the contents. It returns an array of record objects created if the call succeeded.
+
+Create a new file, `createItem.js` in the `/pages/api` folder, and add the following code.
+
+```js
+import { table, getMinifiedItem } from '../../utils/Airtable'
+
+export default async (req, res) => {
+  const { item } = req.body
+  try {
+    const newRecords = await table.create([{ fields: { item } }])
+    res.status(200).json(getMinifiedItem(newRecords[0]))
+  } catch (error) {
+    console.log(error)
+    res.status(500).json({ msg: 'Something went wrong! 😕' })
+  }
+}
+```
+
+You can use [Postman](https://www.postman.com/) or something similar to send a request and test the endpoint.
+
+### Update a record
+To update records, you can use the `update` or `replace` method. If you want to update a single field of a record, use the `update` method, or if you're going to replace it with a new record, use `replace` method. The `update` method is very similar to `create` method. It takes an array of `id`s and `fields` up to `10` records and returns the array of updated records.
+
+Create a new file `updateItem.js` in the `/pages/api` folder and add the following code.
+
+```js
+import { table, getMinifiedItem } from '../../utils/Airtable'
+
+export default async (req, res) => {
+	const { id, fields } = req.body
+	try {
+		const updatedRecords = await table.update([{ id, fields }])
+		res.status(200).json(getMinifiedItem(updatedRecords[0]))
+	} catch (error) {
+		console.log(error)
+		res.status(500).json({ msg: 'Something went wrong! 😕' })
+	}
+}
+```
+
+Here, you are retrieving the record corresponding to the id and update the fields with new values. You can test the endpoint by sending a request to the API using Postman.
+
+### Delete a record
+You can delete a record using the `destroy` method. It takes an array of `id`s of the records you want to delete. You can also set the first parameter to a record ID to delete a single record. It returns the deleted record.
+
+Create a new file `deleteItem.js` in the `/pages/api` folder and add the following code.
+
+```js
+import { table } from '../../utils/Airtable'
+
+export default async (req, res) => {
+	const { id } = req.body
+	try {
+		const deletedRecords = await table.destroy([id])
+		res.status(200).json(deletedRecords)
+	} catch (error) {
+		console.log(error)
+		res.status(500).json({ msg: 'Something went wrong! 😕' })
+	}
+}
+```
+
```

Comment:
Consider adding an image that contains the delete request
One where the body of the request is visible